cartofacade/cgo_api.go

```diff
@@ -0,0 +1,456 @@
+// Package cartofacade provides an api to call into c code
+//
+//nolint:lll
+package cartofacade
+
+/*
+	#cgo CFLAGS: -I../viam-cartographer/src/carto_facade
+
+	// the libraries that need to be linked can be dermined by checking line 258 of the build.ninja file that is autogenerated during make build
+  	#cgo LDFLAGS: -L../viam-cartographer/build -L../viam-cartographer/cartographer/build -lcartographer  -lviam-cartographer  -labsl_raw_hash_set -labsl_hash  -labsl_str_format_internal  -labsl_synchronization    -labsl_strings  -labsl_throw_delegate -lcairo -llua -lstdc++ -lceres -lprotobuf -lglog -lboost_filesystem -lboost_iostreams -lpcl_io -lpcl_common
+
+	#include "../viam-cartographer/src/carto_facade/carto_facade.h"
+
+	viam_carto_lib* alloc_viam_carto_lib() { return (viam_carto_lib*) malloc(sizeof(viam_carto_lib)); }
+	void free_alloc_viam_carto_lib(viam_carto_lib* p) { free(p); }
+
+	viam_carto* alloc_viam_carto() { return (viam_carto*) malloc(sizeof(viam_carto)); }
+	void free_alloc_viam_carto(viam_carto* p) { free(p); }
+
+	bstring* alloc_bstring_array(size_t len) { return (bstring*) malloc(len * sizeof(bstring)); }
+	void free_bstring_array(bstring* p, int len) {
```

Comment:
to make the  length type symmetrical with alloc
```suggestion
	void free_bstring_array(bstring* p, size_t len) {
```

Reply:
Also, I'm very sorry to say that `bdestroy` is not a void function & in fact returns errors if called with invalid inputs.
Could you please propagate the error returned from bdestroy if it returns `BSTR_ERR` & `BSTR_OK` on success & check the error in go?

Reply:
`alloc_bstring_array takes ``size_t` because malloc requires it. `free_bstring_array` needs to take an int so that we can iterate with it. Is that okay?